src/ArrowTypes/src/ArrowTypes.jl

```diff
@@ -348,6 +348,8 @@ function default(::Type{A}) where {A <: AbstractVector{T}} where {T}
     return a
 end
 
+default(::Type{SubArray{T,N,P,I,L}}) where {T,N,P,I,L} = default(P)
```

Comment:
```suggestion
default(::Type{SubArray{T,N,P,I,L}}) where {T,N,P,I,L} = view(default(P), 0:-1)
```
The issue here is `default(P)` returns a `Vector`, but we want a SubArray, so my suggestion seems to solve it for me locally

Reply:
Good suggestion, thanks.